functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/jobs/JobActivityHistorySpec.groovy

```diff
@@ -0,0 +1,82 @@
+package org.rundeck.tests.functional.selenium.jobs
+
+import org.openqa.selenium.By
+import org.openqa.selenium.WebElement
+import org.rundeck.util.annotations.SeleniumCoreTest
+import org.rundeck.util.container.SeleniumBase
+import org.rundeck.util.container.SeleniumContext
+import org.rundeck.util.gui.pages.login.LoginPage
+import org.rundeck.util.gui.pages.home.HomePage
+import org.rundeck.util.gui.pages.jobs.JobCreatePage
+import org.rundeck.util.gui.pages.execution.ExecutionShowPage
+import org.rundeck.util.gui.pages.execution.HtmlRenderedOutputPage
+import org.rundeck.util.gui.pages.jobs.JobListPage
+import org.rundeck.util.gui.pages.activity.ActivityPage
+import org.rundeck.util.gui.pages.jobs.JobTab
+import org.rundeck.util.gui.pages.jobs.JobShowPage
+import spock.lang.Stepwise
+@SeleniumCoreTest
+@Stepwise
+class JobActivityHistorySpec extends SeleniumBase {
+    def setupSpec() {
+        setupProject(SELENIUM_BASIC_PROJECT)
+    }
+    def setup() {
+        def loginPage = go LoginPage
+        loginPage.login(TEST_USER, TEST_PASS)
+        println "Login successful"
+    }
+    def "create job has basic fields"() {
+        when:
+        def jobCreatePage = go JobCreatePage, SELENIUM_BASIC_PROJECT
+        then:
+        jobCreatePage.validatePage()
+        jobCreatePage.jobNameInput
+        jobCreatePage.groupPathInput
+        jobCreatePage.descriptionTextarea
+        println "job created successfully"
+    }
+    def "run job and validate execution"() {
+        def executionShowPage = page ExecutionShowPage
+        def jobShowPage = go JobShowPage, SELENIUM_BASIC_PROJECT
+        when: "Navigate to the Job Show Page and run the job"
+        println "Navigating to Job Show Page"
+        jobShowPage.validatePage()
+        println "Job Show Page validated"
+        def runJobButton = driver.findElement(By.cssSelector(".btn.btn-success.btn-simple.btn-hover.btn-xs.act_execute_job"))
```

Comment:
Isolate css selectors to the `JobShowPage`.
Use "helpers" from the `JobShowPage` e.g. `runJobLink`

Reply:
> Isolate css selectors to the `JobShowPage`. Use "helpers" from the `JobShowPage` e.g. `runJobLink`

I have not  implemented  this yet as I wanted to confirm it first.